crates/dojo-world/src/migration/strategy.rs

```diff
@@ -145,7 +138,16 @@ pub fn prepare_for_migration(
         world.contract_address = generated_world_address;
     }
 
-    Ok(MigrationStrategy { world_address, world, base, contracts, models, metadata })
+    // If world address is not provided, then we expect the world to be migrated.
+    // TODO: see if there are any case where world_address is not provided and `world` is none
```

Comment:
We should always be able to compute the world address from the local information. We should ensure this is something true all along the migration process.

Reply:
yup make sense, i just put a TODO there to verify this constraint. But whenever  world_address is not provided we try to deploy it. will remove the TODO.